File: accesshqWebTesting/src/test/java/com/accesshq/model/Planet.java
package com.accesshq.model;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.text.NumberFormat;
import java.text.ParseException;
import java.util.ArrayList;
import java.util.List;

public class Planet {

    private final WebElement planetElement;

    public Planet(WebElement planetElement) {
        this.planetElement = planetElement;
    }

    public String getName() {
        return planetElement.findElement(By.className("name")).getText();
    }

    public Double getRadius() throws ParseException {
        String text = planetElement.findElement(By.className("radius")).getText();
        return NumberFormat.getInstance().parse(text).doubleValue();
    }

    public Double getDistance() throws ParseException {
        String text = planetElement.findElement(By.className("distance")).getText();
        return NumberFormat.getInstance().parse(text).doubleValue();
    }

    public void clickExplore() {
        planetElement.findElement(By.tagName("button")).click();
    }
}

File: accesshqCsharp/CodeCampCsharp/CodeCampCsharp/model/Planet.cs
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CodeCampCsharp.model
{
    public class Planet
    {
        private WebElement planetElement;

        public Planet(WebElement PlanetElement)
        {
            this.planetElement = PlanetElement;
        }

        public string GetName()
        {
            return planetElement.FindElement(By.ClassName("name")).Text;
        }

        public double GetRadius()
        {
            string text = planetElement.FindElement(By.ClassName("radius")).Text;
            string numberPart = text.Split().Last();
            double num;
            bool validNum = double.TryParse(numberPart, NumberStyles.Any, CultureInfo.InvariantCulture, out num);
            /*            double doubleValue = Convert.ToDouble(text);*/
            /*            return validNum;*/
            if (validNum)
                return num;
            else
                throw new TypeAccessException("wrong type");

        }

        public double GetDistance()
        {
            string Text = planetElement.FindElement(By.ClassName("distance")).Text;
            return double.Parse(Text);
        }

        public void ClickExplore()
        {
            planetElement.FindElement(By.TagName("button")).Click();
        }

    }
}
